staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -454,7 +455,6 @@ func (r *Request) Body(obj interface{}) *Request {
 		}
```

Comment:
for the string/[]byte/reader cases above, I could see setting Content-Type if there's no existing content-type header and len(r.c.content.ContentType) > 0